### DISCUSSION TITLE
Extract Hamiltonian parameters from DFT+DMFT output

### DISCUSSION THREAD

**Discussion by AQuantumGuy** [TRIQS/dft_tools#241] [Q&A]
Dear TRIQS community, my question is regarding my DFT+DMFT calculations (using Quantum Espresso, Wannier90 and TRIQS). I've performed these simulations and now I wanted to perfrom Quantum algorithms simulations, for this I need to extract the Hamiltonian parameters from my output. In specific, I wanted the values for:

- U (float) – Coulomb repulsion intensity.
- mu (float) – Chemical potential.
- V (np.ndarray) – Tunneling energies. This vector has the same size as the number of bath mode.
- epsilon (np.ndarray) – Bath modes energies. This vector has the same size as the number of bath mode.
- t_mat (np.ndarray) – Hopping matrix (n_sites x n_sites).

U and mu are given by myself in the DMFT input file in TRQIS, I wanted to ask if you can guide me on finding the rest of the parameters I've mentioned in my output files. Could you please let me know where to find?

Thanks in advance!

**Comment by the-hampel** [comment]
Dear @AQuantumGuy, 
I moved your question to dft_tools. 

I cannot judge from a physics point of view of this makes sense but I can outline how you can extract these parameters you are asking for.
* `U` is indeed an input parameter of your calculation
* `mu` you can extract at the end of your DMFT calculation (this is not an input parameter, but calculated new in each DMFT step)
* `t_mat` can be extracted from the input. When using Wannier90 the easiest way to extract this is from the `seed_hr.dat` file created from wannier90. You can load this file via: 
```
from triqs.lattice.utils import TB_from_wannier90, k_space_path
import matplotlib.pyplot as plt

svo_tb = TB_from_wannier90(seed='svo_t2g' ,path='./')
svo_tb.hoppings[(0,0,0)] # here the tuple ( ... ) is any R vector and will give back a norb x norb matrix 
```
* `V` and `epsilon` can be obtained by loading the final Weiss field of your DMFT calculation `G0_iw` and fitting a dicretized bath coupling to it via https://triqs.github.io/triqs/unstable/documentation/python_api/triqs.gf.tools.discretize_bath.html#triqs-gf-tools-discretize-bath 

Best,
Alex

**Reply by AQuantumGuy** [reply]
Hi Alex, many thanks for your reply and clarifications. I'll make sure to have a look into that!

I want maybe your input on this. I was planning to perform DFT+DMFT simulations and then use quantum algorithms (like VQE) and perform calculations of ground energy and then Green's function (so to then compare it to DFT+DMFT simulations)

Below is a step-by-step list from the initial DFT calculation to the final comparison of Green's functions using a quantum computer. Could you let me know if this is the right approach to perform DFT+DMFT and solving the Green's functions on a Quantum Simulator? As in a physical sense?

- Perform a Density Functional Theory (DFT) calculation using Quantum Espresso (QE) to obtain the electronic structure of the material.
- Use Wannier90 to construct the Wannier functions and obtain a tight-binding model.
- Employ TRIQS for the Dynamical Mean-Field Theory (DMFT) calculations, which will involve the following steps:
      a. Construct a local impurity problem using the DFT output and the tight-binding model.
      b. Set up and solve the Anderson impurity model (AIM) using TRIQS, for example, with the cthyb solver.
      c. Perform the DMFT self-consistency loop, which involves calculating the Green's functions and updating the local self-energy.
- Extract the impurity model Hamiltonian parameters (like those mentioned in my question) from the TRIQS output.
- Translate the Hamiltonian into a qubit representation, for example, by applying the Jordan-Wigner transformation.
- Construct a variational quantum circuit (ansatz) to approximate the ground state of the qubit Hamiltonian
- Optimize the parameters of the variational quantum circuit using a classical optimizer to minimize the expectation value of the Hamiltonian within the VQE algorithm.
- Run the VQE algorithm on a quantum simulator, using a quantum computing platform
- Compute the Green's function using the approximate ground state obtained from the VQE algorithm by applying a simple quantum circuit.
- Analyze the results and compare the Green's function obtained from the quantum computation with the original DFT+DMFT results.

What are your thoughts on this? (as in general sense)

**Reply by the-hampel** [reply]
Hi @AQuantumGuy ,

I think the first four points make sense to me. However, I have little experience with VQE algorithm's and mapping the AIM to quantum computers. Hence, I cannot give you any advice for the remainder of the list. I know that people are generally working on this and that this is current research. 

I would encourage you to check arxiv for papers that are doing these kind of mappings and try to reach out to one of these authors. Within triqs we do not have any tutorial or so on this matter. 

Maybe a different user notices this discussion and has some insights.

Best,
Alex
